test/identity-provider.test.js

```diff
@@ -4,99 +4,79 @@ const assert = require('assert')
 const path = require('path')
 const rmrf = require('rimraf')
 const Keystore = require('orbit-db-keystore')
-const IdentityProvider = require('../src/identity-provider')
+const Identities = require('../src/identities')
 const Identity = require('../src/identity')
-const { Wallet } = require('ethers')
 const savedKeysPath = path.resolve('./test/fixtures/keys')
-const testKeysPath = path.resolve('./test/keys')
+const keypath = path.resolve('./test/keys')
 let keystore
 
+const type = 'orbitdb'
+
 describe('Identity Provider', function () {
-  const type = 'orbitdb'
-  const identitySignerFn = async (id, data) => {
-    const key = await keystore.getKey(id)
-    return keystore.sign(key, data)
-  }
-
-  const identityVerifierFn = async (identity) => {
-    return keystore.verify(identity.signatures.publicKey, identity.publicKey, identity.publicKey + identity.signatures.id)
-  }
-
-  before(() => {
-    // Make sure we don't use previous test keys
-    rmrf.sync(testKeysPath)
-    keystore = Keystore.create(testKeysPath)
+  before(async () => {
+    rmrf.sync(keypath)
+    keystore = Keystore.create(keypath)
   })
 
-  after(() => {
+  after(async () => {
     // Remove stored keys
-    rmrf.sync(testKeysPath)
-  })
-
-  describe('constructor', () => {
-    it('throws an error if keystore is not given as a constructor argument', async () => {
-      let err
-      let identity
-
-      try {
-        identity = new IdentityProvider()
-      } catch (e) {
-        err = e
-      }
-
-      assert.strictEqual(identity, undefined)
-      assert.strictEqual(err.message, 'Keystore is required')
-    })
+    rmrf.sync(keypath)
   })
 
   describe('create an identity', () => {
     describe('create a new identity', () => {
-      const id = '0x01234567890abcdefghijklmnopqrstuvwxyz'
-      let identity
+      let id = 'A'
+      let identity, externalPublicKey
```

Comment:
Bearing in mind that the `id` can also be something else than public key, perhaps rename `externalPublicKey` to `externalId`?